src/GraphQL.SystemTextJson/ObjectDictionaryConverter.cs

```diff
@@ -0,0 +1,223 @@
+        {
+            foreach (JsonElement item in value.EnumerateArray())
+            {
+                switch (item.ValueKind)
+                {
+                    case JsonValueKind.Object:
+                        yield return ReadDictionary(item);
+                        break;
+                    case JsonValueKind.Number:
+                        yield return ReadNumber(item);
+                        break;
+                    case JsonValueKind.True:
+                    case JsonValueKind.False:
+                        yield return item.GetBoolean();
+                        break;
+                    case JsonValueKind.String:
+                        yield return item.GetString();
+                        break;
+                    case JsonValueKind.Null:
+                        yield return null;
+                        break;
+                    case JsonValueKind.Array:
+                        yield return ReadArray(item).ToList();
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Unexpected value kind: {item.ValueKind}");
+                }
+            }
+        }
+
+        private object ReadNumber(JsonElement value)
+        {
+            if (value.TryGetInt32(out var i))
+                return i;
+            else if (value.TryGetInt64(out var l))
+                return l;
+            //else if (value.TryGetUInt64(out var ui))
+            //    resultValue = ui;
+            else if (BigInteger.TryParse(value.GetRawText(), out var bi))
+                return bi;
+            else if (value.TryGetDouble(out var d))
+                return d;
+            else if (value.TryGetDecimal(out var dd))
+                return dd;
+
+            throw new NotImplementedException($"Unexpected Number value. Raw text was: {value.GetRawText()}");
+        }
+
+        private void WriteDictionary(Utf8JsonWriter writer, Dictionary<string, object> dictionary,
+            JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+
+            foreach (var entry in dictionary)
+            {
+                var value = entry.Value;
+
+                if (value == null)
+                {
+                    writer.WriteNull(entry.Key);
+                    continue;
+                }
+
+                WriteProperty(writer, options, value, entry.Key);
+            }
+
+            writer.WriteEndObject();
+        }
+
+        private void WriteProperty(Utf8JsonWriter writer, JsonSerializerOptions options, object value, string key)
+        {
+            switch (value)
+            {
+                case int intValue:
+                    writer.WriteNumber(key, intValue);
+                    break;
+                case double doubleValue:
+                    writer.WriteNumber(key, doubleValue);
+                    break;
+                case decimal decimalValue:
```

Comment:
float, ulong, uint, etc cases ?

Reply:
Write methods were removed. This is no longer relevant.